src/inputs/kafka_input.rs

```diff
@@ -267,6 +267,7 @@ impl KafkaInput {
 }
 
 impl InputReader<TdPyAny> for KafkaInput {
+    #[tracing::instrument(name = "kafka_input", level = "trace", skip_all)]
```

Comment:
Do these macros automatically cause the span resulting from a call to this function to be grouped under the currently open "step" and "epoch"? Or how is the hierarchy defined?

Reply:
Yes, it's basically time related, if the instrumented function is called while another span is open, it will be considered a child of that span.